src/stories/NewButton.stories.tsx

```diff
@@ -0,0 +1,136 @@
+import React, {useState} from 'react'
+import Button, {ButtonProps} from '../NewButton'
+import {BaseStyles, ThemeProvider} from '..'
+import {Meta} from '@storybook/react'
+import {XIcon, SearchIcon, EyeIcon, EyeClosedIcon} from '@primer/octicons-react'
+
+export default {
+  title: 'Composite components/New Button',
+
+  decorators: [
+    Story => {
+      return (
+        <ThemeProvider>
+          <BaseStyles>
+            <Story />
+          </BaseStyles>
+        </ThemeProvider>
+      )
+    }
+  ],
+  argTypes: {
+    size: {
+      control: {
+        type: 'radio',
+        options: ['small', 'medium', 'large']
+      }
+    }
+  }
+} as Meta
+
+export const defaultButton = ({size = 'medium', ...args}: ButtonProps) => {
+  return (
+    <Button size={size} {...args}>
+      Default
+    </Button>
+  )
+}
+
+export const primaryButton = (args: ButtonProps) => {
+  return (
+    <Button {...args} variant="primary">
+      Primary
+    </Button>
+  )
+}
+
+export const dangerButton = (args: ButtonProps) => {
+  return (
+    <Button {...args} variant="danger">
+      Danger
+    </Button>
+  )
+}
+
+export const invisibleButton = (args: ButtonProps) => {
+  return (
+    <Button {...args} variant="invisible">
+      Invisible
+    </Button>
+  )
+}
+
+export const iconBeforeButton = (args: ButtonProps) => {
+  return (
+    <Button icon={() => <SearchIcon />} {...args}>
```

Comment:
Do we want to allow folks to pass props to the icon itself (size/color) or is that always dictated by the Button size+variant. 

If it's the latter, we might want to promote usage like this in the docs: `Button icon={SearchIcon}>`

Reply:
TBH, I went for this API because thats what I was able to type correctly. You are right though, we don't want consumers to provide styles. We want to add a few size and a11y params to the icon though. Like suggested by @langermank in the comment below.

Reply:
@pksjce - we could just wrap the SVG in a container that has a width and height. Then, if somebody tries to pass an icon size that's too large, it would overflow the container and look obviously broken. If we want to make sure the icon never overflows the container, we could even do something like `.buttonContainer > svg { max-width: 100%; }`

Reply:
Convo with @ashygee that icon size won't change between buttons, so we don't need to accept that prop. 
The a11y strategy is to hide icons since they are only a visual artifact. My discussion with @langermank is to mandate button having associated text always. Though I'm still exploring how this will work in iconOnly scenarios. 
Will we allow consumers to change other icon properties? like color and such? I don't think we need to do that. Everything should be globally themeable and not locally customisable right?